bee-ledger/src/types/receipt.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::types::{error::Error, TreasuryOutput};
+
+use bee_common::packable::{Packable, Read, Write};
+use bee_message::{
+    constants::IOTA_SUPPLY,
+    input::Input,
+    milestone::MilestoneIndex,
+    output::Output,
+    payload::{receipt::ReceiptPayload, Payload},
+};
+
+/// A type that wraps a receipt and the index of the milestone in which it was included.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Receipt {
+    inner: ReceiptPayload,
+    included_in: MilestoneIndex,
+}
+
+impl Receipt {
+    /// Creates a new `Receipt`.
+    pub fn new(inner: ReceiptPayload, included_in: MilestoneIndex) -> Self {
+        Self { inner, included_in }
+    }
+
+    /// Returns the inner receipt of the `Receipt`.
+    pub fn inner(&self) -> &ReceiptPayload {
```

Comment:
I'ld suggest renaming this API. I'ld only name it `inner` if we'ld also deref to this "inner" type.

Reply:
I think it's fine